langchain/agents/tests/json.test.ts

```diff
@@ -0,0 +1,71 @@
+import { test, expect } from "@jest/globals";
+import { JsonListKeysTool, JsonSpec, JsonGetValueTool } from "../tools/json";
+
+test("JsonListKeysTool", async () => {
+  const jsonSpec = new JsonSpec({
+    foo: "bar",
+    baz: { test: { foo: [1, 2, 3], qux: [{ x: 1, y: 2, z: 3 }, { a: 1 }] } },
+  });
+  const jsonListKeysTool = new JsonListKeysTool(jsonSpec);
+  expect(await jsonListKeysTool.call("")).toBe("foo, baz");
+  expect(await jsonListKeysTool.call("/foo")).toContain("not a dictionary");
+  expect(await jsonListKeysTool.call("/baz")).toBe("test");
+  expect(await jsonListKeysTool.call("/baz/test")).toBe("foo, qux");
+  expect(await jsonListKeysTool.call("/baz/test/foo")).toContain(
+    "not a dictionary"
+  );
+  expect(await jsonListKeysTool.call("/baz/test/foo/0")).toContain(
+    "not a dictionary"
+  );
+  expect(await jsonListKeysTool.call("/baz/test/qux")).toContain(
```

Comment:
Arguable here it could return `0, 1` but not sure how useful that is in this case